plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -59,34 +334,169 @@ constraints, affinity and anti-affinity specifications, data locality, inter-wor
 interference, deadlines, and so on. Workload-specific requirements will be exposed
 through the API as necessary.`,
 		Run: func(cmd *cobra.Command, args []string) {
+			verflag.PrintAndExitIfRequested()
+			cmdutil.CheckErr(opts.Complete())
+			cmdutil.CheckErr(opts.Validate(args))
+			cmdutil.CheckErr(opts.Run())
 		},
 	}
 
+	opts.config, err = opts.ApplyDefaults(opts.config)
+	if err != nil {
+		glog.Fatalf("unable to apply config defaults: %v", err)
+	}
+
+	flags := cmd.Flags()
+	AddFlags(opts, flags)
+
+	cmd.MarkFlagFilename("config", "yaml", "yml", "json")
+
 	return cmd
 }
 
-// Run runs the specified SchedulerServer.  This should never exit.
-func Run(s *options.SchedulerServer) error {
-	// To help debugging, immediately log version
-	glog.Infof("Version: %+v", version.Get())
+// SchedulerServer represents all the parameters required to start the
+// Kubernetes scheduler server.
+type SchedulerServer struct {
+	SchedulerName                  string
+	Client                         clientset.Interface
+	AlgorithmSource                componentconfig.SchedulerAlgorithmSource
+	HardPodAffinitySymmetricWeight int
+	EventClient                    v1core.EventsGetter
+	Recorder                       record.EventRecorder
+	Broadcaster                    record.EventBroadcaster
+	// LeaderElection is optional.
+	LeaderElection *leaderelection.LeaderElectionConfig
+	// HealthzServer is optional.
+	HealthzServer *http.Server
+	// MetricsServer is optional.
+	MetricsServer *http.Server
+}
+
+// NewSchedulerServer creates a runnable SchedulerServer from configuration.
+func NewSchedulerServer(config *componentconfig.KubeSchedulerConfiguration, master string) (*SchedulerServer, error) {
```

Comment:
Please break this function into smaller functions. It is too long.

Reply:
Decomposed.